pkg/flow/internal/controller/loader.go

```diff
@@ -217,6 +217,17 @@ func (l *Loader) populateConfigBlockNodes(g *dag.Graph, configBlocks []*ast.Bloc
 		g.Add(c)
 	}
 
+	// If a logging config block is not provided, use the defaults.
```

Comment:
This needs to be done for he tracing block too. 

Also, I think this is OK for now but I'm worried about how maintainable it is. Can we add a TODO to try to identify a solution which we're less likely to accidentally break?

Reply:
(Longer term, I think the logging/tracing config nodes always exist in the DAG, even when the block isn't provided in the config file. That way we establish a pattern of having config nodes which always exist and always have some kind of evaluation logic)

Reply:
> I'm open to adding it to the DAG if it's missing instead if we prefer that approach. 

IMO it'll be a little more code but I think it's preferable just from the perspective of keeping logging specific code isolated, and not having the loader be aware of specific subsystems. 

> What do you think about the opts caching in the logger?

I don't really have an opinion about that one, I think it's fine 👍

Reply:
So I started implementing this and I may be going down the wrong path or an unexpected rabbit hole.

I think a couple things are true
- We need to do something in `loader.go` `populateConfigBlockNodes` to add the node after it has finished adding other nodes so we aren't totally free of custom logic for this type of situation in the loader.
- Creating a block without real values on it for NamePos, LabelPos, etc feels kind of wrong.
- I'm concerned what will happen if we walk the graph on a fake node how it will behave but maybe it's ok


Prototype I was working on...

loader.go populateConfigBlockNodes replace change in this PR with...
```golang
        If a logging config block is not provided add it to the graph.
	if _, ok := blockMap[loggingBlockID]; !ok && !l.isModule() {
		c, newConfigNodeDiags := NewDefaultLoggingConfigNode(l.globals, l.isModule())
		diags = append(diags, newConfigNodeDiags...)
		g.Add(c)
	}
```

config_logging.go
```golang
func NewDefaultLoggingConfigNode(globals ComponentGlobals, isInModule bool) (*LoggingConfigNode, diag.Diagnostics) {
        // Fill all this in with logging defaults
        block := &ast.BlockStmt{
		Name:      []string{},
		NamePos:   token.Pos{},
		Label:     "",
		LabelPos:  token.Pos{},
		Body:      []ast.Stmt{},
		LCurlyPos: token.Pos{},
		RCurlyPos: token.Pos{},
	}

	return NewLoggingConfigNode(block, globals, isInModule)
}
```

Were you thinking of something different or more drastic than this approach? I can keep thinking on it too.

Reply:
I could make the case that populateConfigBlockNodes belongs in config.go and populateComponentNodes in component.go too (pass in the 2 things we need from `l *Loader`). Thoughts?

Reply:
we talked offline - I'm going to implement with a nil block on the logging node and see what it looks like

Reply:
I added both logging and tracing back into the dag while I was at it. This should make it easier to similarly propagate the tracing for modules @mattdurham

Reply:
This should be all set pending another review of the specific implementation details